Date: 2010-09-02 13:47:33 -0700
Author: awinterz
Description:

commit Brad Hards regression test for un-escaped characters in TZID.

diff -Naur libical-0.46/src/test/regression.c libical-0.46-new/src/test/regression.c
--- libical-0.46/src/test/regression.c	2010-08-30 13:05:03.000000000 -0700
+++ libical-0.46-new/src/test/regression.c	2010-11-30 15:24:13.436857765 -0800
@@ -1277,6 +1277,24 @@
 #endif
 
 
+void test_tzid_escape(){
+    icalparameter *tzid;
+    icalproperty *prop;
+
+    tzid = icalparameter_new_tzid("Timezone\nwith a newline");
+    prop = icalproperty_new_dtstart(icaltime_from_day_of_year(26, 2009));
+    icalproperty_add_parameter(prop, tzid);
+
+    if (VERBOSE)
+      printf("%s\n",icalproperty_as_ical_string(prop));
+
+    is("test encoding of 'Timezone\\nwith a newline'",
+       icalproperty_as_ical_string(prop), "DTSTART;VALUE=DATE,TZID=Timezone\\nwith a newline:20090126");
+
+    icalproperty_free(prop);
+}
+
+
 void test_requeststat()
 {
     icalproperty *p;
@@ -3770,6 +3788,7 @@
     test_run("Test Convenience", test_convenience, do_test, do_header);
     test_run("Test classify ", test_classify, do_test, do_header);
     test_run("Test Iterators", test_iterators, do_test, do_header);
+    test_run("Test TZID escaping", test_tzid_escape, do_test, do_header);
     test_run("Test strings", test_strings, do_test, do_header);
 #ifdef INVALID_TEST
     test_run("Test TZID escaping", test_tzid_escape, do_test, do_header);
